src/basic/cgroup-util.c

```diff
@@ -68,6 +68,58 @@ int cg_cgroupid_open(int cgroupfs_fd, uint64_t id) {
         return RET_NERRNO(open_by_handle_at(cgroupfs_fd, &fh.file_handle, O_DIRECTORY|O_CLOEXEC));
 }
 
+int cg_path_from_cgroupid(int cgroupfs_fd, uint64_t id, char **ret) {
+        _cleanup_close_ int cgfd = -EBADF;
+        int r;
+
+        cgfd = cg_cgroupid_open(cgroupfs_fd, id);
+        if (cgfd < 0)
+                return cgfd;
+
+        _cleanup_free_ char *path = NULL;
+
+        r = fd_get_path(cgfd, &path);
```

Comment:
BTW, IIRC, we do not have any rules about variable declaration in a middle of function, but I am not a fan of inserting an empty line between the declaration and its use. At least I and Lennart do not insert such.

Reply:
Dunno. Unless the initialization is a function call, I really dislike placing stuff tightly together...